### PR TITLE
Address integration test and build issues

### PR DESCRIPTION
**Pull request by bgeltz**
- Relates to #3386.

### CODE REVIEW

File: integration/service/Makefile.mk [hunk 1]
```diff
@@ -2,61 +2,70 @@
...
 
-noinst_PROGRAMS += test/test_batch_server \
-                  test/test_batch_interface \
-                  test/test_invalid_values \
-                  test/test_batch_perf \
-                  #end
+noinst_PROGRAMS += service/test/test_batch_interface \
+		   service/test/test_invalid_values \
+		   service/test/test_batch_perf \
+		   #end
```
Comment (bgeltz) [inline integration/service/Makefile.mk]: wsp

File: integration/service/Makefile.mk [hunk 2]
```diff
@@ -2,61 +2,70 @@
...
 
-test_test_batch_server_SOURCES = test/test_batch_server.cpp
-test_test_batch_server_LDADD = libgeopmd.la
+if ENABLE_GEOPMD_LOCAL
+  # This test depends on non-installed libgeopmd headers
+  noinst_PROGRAMS += service/test/test_batch_server \
+		     #end
```
Comment (bgeltz) [inline integration/service/Makefile.mk]: wsp

File: libgeopm/test/Makefile.mk [hunk 3]
```diff
@@ -3,15 +3,19 @@
...
     check_PROGRAMS += test/geopm_mpi_test_api
 endif
 
 TEST_LOG_DRIVER = env AM_TAP_AWK='$(AWK)' $(SHELL) \
 				  $(top_srcdir)/build-aux/tap-driver.sh
-TESTS += test/geopm_test.test \
-         # end
+
+AM_TESTS_ENVIRONMENT = geopm_test_path='$(top_builddir)'/test/geopm_test; \
+		       export geopm_test_path;
```
Comment (bgeltz) [inline libgeopm/test/Makefile.mk]: wsp

File: libgeopmd/test/Makefile.mk [hunk 4]
```diff
@@ -5,18 +5,21 @@
...
 test_isadmin_SOURCES = test/isadmin.cpp
 test_isadmin_LDADD = libgeopmd.la
 
 TEST_LOG_DRIVER = env AM_TAP_AWK='$(AWK)' $(SHELL) \
                   $(top_srcdir)/build-aux/tap-driver.sh
 
-TESTS += test/geopm_test.test \
-         # end
+AM_TESTS_ENVIRONMENT = geopm_test_path='$(top_builddir)'/test/geopm_test; \
+		       export geopm_test_path;
```
Comment (bgeltz) [inline libgeopmd/test/Makefile.mk:18]: wsp

**Review by cmcantalupo — APPROVED** [review]

**Pull request merged**